python/pyspark/sql/connect/dataframe.py

```diff
@@ -1631,6 +1632,47 @@ def inputFiles(self) -> List[str]:
         query = self._plan.to_proto(self._session.client)
         return self._session.client._analyze(query).input_files
 
+    def to(self, schema: Union[DataType, str]) -> "DataFrame":
```

Comment:
PySpark API does not support this as a string. Let's get rid of string support for now.

Reply:
+1

Reply:
OK

Reply:
yes we should have StructType in `pyspark_types_to_proto_types`.

Reply:
I have created https://github.com/apache/spark/pull/39103 let `pyspark_types_to_proto_types` support `StructType`.